lib/Migration/Version1120Date20220412111833.php

```diff
@@ -0,0 +1,134 @@
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Type;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version1120Date20220412111833 extends SimpleMigrationStep {
+	/** @var IDBConnection $connection */
+	private $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		// Truncate all tables that will get a new index later
+		$qb = $this->connection->getQueryBuilder();
+		$qb->delete('mail_recipients');
+		$qb->execute();
+
+		// Truncate and change primary key type for messages table
+		$qb1 = $this->connection->getQueryBuilder();
+		$qb1->delete('mail_messages');
+		$qb1->execute();
+
+		// Truncate message_tags table
+		$qb2 = $this->connection->getQueryBuilder();
+		$qb2->delete('mail_message_tags');
+		$qb2->execute();
+
+		// unset all locks for the mailboxes table
+		$qb3 = $this->connection->getQueryBuilder();
+		$qb3->update('mail_mailboxes')
+			->set('sync_new_lock', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL))
+			->set('sync_changed_lock', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL))
+			->set('sync_vanished_lock', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL))
+			->set('sync_new_token', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL))
+			->set('sync_changed_token', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL))
+			->set('sync_vanished_token', $qb3->createNamedParameter(null, IQueryBuilder::PARAM_NULL));
+		$qb3->execute();
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$schema = $schemaClosure();
+
+		// Change primary column to bigint
+		$messagesTable = $schema->getTable('mail_messages');
+		$messagesTable->changeColumn('id', [
+			'type' => Type::getType('bigint'),
```

Comment:
shouldn't this be just int and the `length` makes it a big one?

Reply:
I guess that can work. Let me try it.